BUILDING_CALICO.md

```diff
@@ -0,0 +1,292 @@
+of multiple sub components each indivudally versioned.  To determine which
+particular tag (version) of code to checkout for each repo, consult the Releases page of the
+appropriate version of the [Calico documentation](http://docs.projectcalico.org)
+
+Define the following environment variables that describe the versions of each of the
+components that you are building (the instruction below use these):
+
+```
+VERSION_TYPHA
+VERSION_FELIX
+VERSION_CNI
+VERSION_CONFD
+VERSION_BIRD
+VERSION_BGP_DAEMON
+VERSION_LIBNETWORK
+VERSION_CALICOCTL
+
+```
+
+For example, the v2.3.0 release of Calico would define the following:
+
+```
+export VERSION_TYPHA=v0.2.0
+export VERSION_FELIX=2.3.0-rc3
+export VERSION_CNI=v1.9.1
+export VERSION_CONFD=v0.12.1-calico0.1.0
+export VERSION_BIRD=v0.3.1
+export VERSION_BGP_DAEMON=v0.2.1
+export VERSION_LIBNETWORK=v1.1.0
+export VERSION_CALICOCTL=v1.3.0-rc1
+```
+
+## Build instructions
+
+The following sections should be followed in order as there are cross dependencies
+between the various components that have been built.
+
+### 1. typha
+
+To build typha, clone the typha repo and checkout the correct tag.
+
+```
+cd $BASEDIR
+git clone git@github.com:projectcalico/typha.git
+cd typha
+git checkout tags/$VERSION_TYPHA
+```
+
+Build the `calico/typha` docker image (which also builds the associated binaries) and
+retag the image with the correct version.
+
+```
+make clean
+make calico/typha
+docker tag calico/typha calico/typha:$VERSION_TYPHA
+```
+
+#### What has this built?
+
+-  The `calico/typha` Docker image
+-  The `bin/calico-typha` binary (included in the Docker image)
+
+### 2. felix (used in the calico/node image)
+
+To build felix, clone the felix repo and checkout the correct tag.
+
+```
+cd $BASEDIR
+git clone git@github.com:projectcalico/felix.git
+cd felix
+git checkout tags/$VERSION_FELIX
+```
+
+Build the `calico/felix` docker image (which also builds the associated binaries) and
+retag the image with the correct version.
+
+```
+make clean
+make calico/felix
+docker tag calico/felix calico/felix:$VERSION_FELIX
+```
+
+#### What has this built?
+
+-  The `calico/felix` Docker image
+-  The `bin/calico-felix` binary (included in the Docker image)
+
+### 3. CNI plugins
+
+To build Calico CNI plugins, clone the felix repo and checkout the correct tag.
+
+```
+cd $BASEDIR
+git clone git@github.com:projectcalico/cni-plugin.git
+cd cni-plugin
+git checkout tags/$VERSION_CNI
+```
+
+Build the `calico/felix` docker image (which also builds the associated binaries) and
+retag the image with the correct version.
+
+```
+make clean
+make docker-image
+docker tag calico/felix calico/felix:$VERSION_FELIX
+```
+
+#### What has this built?
+
+-  The `calico/cni` Docker image
+-  The `dist/calico` binary (included in the Docker image)
+-  The `dist/calico-ipam` binary (included in the Docker image)
+-  The `dist/flannel` binary (included in the Docker image)
```

Comment:
Strictly speaking, the things below this are not built here - they are pulled in from external sources pre-built.